src/com/extendedclip/papi/expansion/checkitem/CheckItemExpansion.java

```diff
@@ -25,1350 +22,1403 @@
+                    + ", checkOffHand="
+                    + checkOffHand
+                    + ", checkEnchantments="
+                    + checkEnchantments
+                    + ", checkEnchanted="
+                    + checkEnchanted
+                    + ", checkNbtStrings="
+                    + checkNbtStrings
+                    + ", checkNbtInts="
+                    + checkNbtInts
+                    + ", isStrict="
+                    + isStrict
+                    + ", hdbItem="
+                    + hdbItem
+                    + ", remove="
+                    + remove
+                    + ", material="
+                    + material
+                    + ", data="
+                    + data
+                    + ", customData="
+                    + customData
+                    + ", amount="
+                    + amount
+                    + ", name="
+                    + name
+                    + ", lore="
+                    + lore
+                    + ", materialString="
+                    + materialString
+                    + ", enchantments="
+                    + enchantments
+                    + ", potionType="
+                    + potionType
+                    + ", checkPotionExtended="
+                    + checkPotionExtended
+                    + ", potionExtended="
+                    + potionExtended
+                    + ", checkPotionUpgraded="
+                    + checkPotionUpgraded
+                    + ", potionUpgraded="
+                    + potionUpgraded
+                    + ", hdbId="
+                    + hdbId
+                    + ", slot="
+                    + slot
+                    + ", nbtStrings="
+                    + nbtStrings
+                    + ", nbtInts="
+                    + nbtInts
+                    + "]";
+        }
+
+        public String getType() {
+            return this.material;
+        }
+
+        protected void setType(String material) {
+            this.material = material.toUpperCase();
+        }
+
+        public short getDurability() {
+            return this.data;
+        }
+
+        protected void setDurability(short durability) {
+            this.data = durability;
+        }
+
+        public int getCustomData() {
+            return this.customData;
+        }
+
+        protected void setCustomData(int customData) {
+            this.customData = customData;
+        }
+
+        public int getAmount() {
+            return this.amount;
+        }
+
+        protected void setAmount(int amount) {
+            this.amount = amount;
+        }
+
+        public String getName() {
+            return this.name;
+        }
+
+        protected void setName(String name) {
+            this.name = name;
+        }
+
+        public String getLore() {
+            return this.lore;
+        }
+
+        public boolean hasLore() {
+            return this.lore != null && !this.lore.isEmpty();
```

Comment:
the `hasLore` method for the item wrapper seems to be returning wrong information.

It returns true/false dependent on if the WRAPPER has lore and if it's not empty.

It should be returning true/false dependent on if `haslore` is in the placeholder or not, nothing to do with the wrapper.

Reply:
In fact, this method could probably just be removed entirely.